Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -1685,6 +1748,23 @@ public class ProgramBuilder {
         // Update type information
         jsTyper.analyze(instr)
 
+        if let op = instr.op as? DifferentialHash, op.allowInnerScope {
```

Comment:
I'm wondering if it'd be nicer to have this (and the logic in `appendDifferentialProbes`) run as a "PostProcessing" step. How about:
- Create a new file "DifferentialFuzzing.swift" in maybe Core/
- Add a class "DifferentialFuzzing" with a static function `instrumentForDifferentialFuzzing(_ program: Program) -> Program` and move this logic in there (create a new ProgramBuilder, append all instructions from the old program, and randomly insert these `DifferentialHash` operations)
- Call that function from fuzzer.execute if we're doing differential fuzzing

WDYT?

Reply:
Another thought, could it make sense to also add a `Return` to every instrumented function (if there isn't one already) to return a random local variable, then insert these hashes for all return values?

Reply:
That's a good point! I think the logic should move into the FuzzEngines, since that's where samples are initially generated. I could imagine something along the lines of:
- Make FuzzEngine a class instead of a protocol
- Add a `doDifferentialFuzzTesting` boolean member and `instrumentForDifferentialTesting` method. The latter contains this logic here
- In FuzzEngine.execute, check if we're doing differential fuzzing, and if so instrument the program

WDYT?

Reply:
DifferentialHash is not an internal operation and should never be emitted into the corpus. It is added within the FuzzEngine code. What do you think of the updated design?